crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1025,6 +1059,113 @@ unsafe impl<'w> SystemParam for DeferredWorld<'w> {
     ) -> Self::Item<'world, 'state> {
         world.into_deferred()
     }
+
+    fn is_exclusive() -> bool {
+        true
+    }
+}
+
+const MUT_WORLD_ERROR: &str = "&mut World requires exclusive access to the \
+    entire world, but a previous system parameter already registered access to \
+    a part of it. Allowing this would break Rust's mutability rules";
+
+// SAFETY: `&mut World` registers exclusive access to the entire world.
+unsafe impl SystemParam for &mut World {
+    type State = ();
+    type Item<'w, 's> = &'w mut World;
+
+    fn init_state(_world: &mut World, system_meta: &mut SystemMeta) -> Self::State {
```

Comment:
Coming from the `apply_deferred` PR, I am quite sure you also have to call `system_meta.set_non_send()` in this method because exclusive systems were non-send as well.
I don't know however why `&World` and `DeferredWorld` do not do that themselves too prior this PR since these have methods that read/write non-send resources.